Plugins~/Src/MeshSyncClient/msPropertyManager.cpp

```diff
@@ -0,0 +1,72 @@
+#include "MeshSyncClient/msPropertyManager.h"
+#include "pch.h"
+#include "MeshSync/SceneGraph/msPropertyInfo.h"
+
+using namespace std;
+
+namespace ms {
+	vector<PropertyInfoPtr> ms::PropertyManager::getAllProperties()
+	{
+		std::unique_lock<std::mutex> lock(m_mutex);
+
+		vector<PropertyInfoPtr> ret;
+
+		for (auto& r : m_records) {
+			ret.push_back(r.propertyInfo);
+		}
+
+		return ret;
+	}
+
+	void PropertyManager::add(PropertyInfoPtr propertyInfo)
+	{
+		std::unique_lock<std::mutex> lock(m_mutex);
+
+		// Make sure the same property doesn't get added twice when objects and their parents are iterated:
+		for (size_t i = 0; i < m_records.size(); ++i) {
```

Comment:
Not sure how a map would help, the 'matches' methhod checks multiple fields, unless we create some kind of hash from that. I don't think it's worth implementing because the number of properties is not that big anyway.